test/validate/mcoe_test.py

```diff
@@ -105,11 +113,11 @@ def test_no_null_rows_mcoe(pudl_out_mcoe, live_dbs, df_name, thresh):
 @pytest.mark.parametrize(
     "df_name,monthly_rows,annual_rows",
     [
-        ("hr_by_unit", 364_920, 30_459),
-        ("hr_by_gen", 557_819, 46_535),
-        ("fuel_cost", 557_819, 46_535),
-        ("capacity_factor", 604_594, 50_448),
-        ("mcoe", 1_077_537, 523_343),
+        ("hr_by_unit", 332_427, 29_624),
+        ("hr_by_gen", 513_394, 45_461),
+        ("fuel_cost", 513_394, 45_461),
+        ("capacity_factor", 4_621_015, 393_750),
+        ("mcoe", 4_625_758, 394_050),
```

Comment:
this is happening bc the `hr_by_unit` relies on the `unit_id_pudl` and the allocated gen table weirdly has slightly less unit id coverage. I'm investigating if I can increase the coverage. but in general we don't expect to have `unit_id_pudl`'s for generators that don't show up in the generator-specific 923 tables so I believe the target here is more or less parity with the old values

Reply:
It turns out the vast majority of these records in the original version (from the generation_eia923 table) had null or 0 net generation. There are 27 monthly records that seem to actually show up in the original table but not the allocated one.

the vast majority of the decrease is because I started dropping null net generation at the end of the aggregation process. I did this primarily for the no-nulls validation tests, but I'm going to remove it and see which outputs actually break with that turned off.